src/app/store/integration-support.service.ts

```diff
@@ -1,27 +1,31 @@
-import { Injectable } from '@angular/core';
-import { Http } from '@angular/http';
+import { Inject, Injectable } from '@angular/core';
+import { Http, Response } from '@angular/http';
 import { Restangular } from 'ngx-restangular';
 import { Observable } from 'rxjs/Observable';
+import { ErrorObservable } from 'rxjs/observable/ErrorObservable';
 
+import { RESTANGULAR_MAPPER } from '../app.module';
 import { ConfigService } from '../config.service';
 import { Action, Connection, Integration } from '../model';
+import { log, getCategory } from '../logging';
 
 @Injectable()
 export class IntegrationSupportService {
-  mapperBaseUrl: string = undefined;
   service: Restangular = undefined;
   filterService: Restangular = undefined;
   metadataService: Restangular = undefined;
+  mapperService: Restangular = undefined;
   configService: ConfigService = undefined;
 
-  constructor(public restangular: Restangular,
+  constructor(@Inject(Restangular) public restangular: Restangular,
+              @Inject(RESTANGULAR_MAPPER) public restangularMapper: Restangular,
```

Comment:
Okay, take out this line and instead inject `injector: Injector`.  This will require adding an `Injector` import from `'@angular/core'` you can add that to line 1 in the file:

```
import { Inject, Injectable, Injector } from '@angular/core';
```

Then in the constructor you can call `injector.get(RESTANGULAR_MAPPER)` which will return the configured restangular instance looks like.  So the full change I have here is:

```
constructor(restangular: Restangular,
              injector: Injector,
              private http: Http,
              private config: ConfigService) {
    this.service = restangular.service('integration-support');
    this.filterService = restangular.service('integrations');
    this.metadataService = restangular.service('connections');
    const restangularMapper = <Restangular> injector.get(RESTANGULAR_MAPPER);
    this.mapperService = restangularMapper.service('java-inspections');
    this.configService = config;
  }
```

Reply:
Thanks @gashcrumb - talk about a workaround.